demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -61,8 +61,11 @@
  * These configuration settings are required to run the basic TLS demo.
  * Throw compilation error if the below configs are not defined.
  */
+#ifndef BROKER_ENDPOINT
+    #error "Please define an MQTT broker endpoint, BROKER_ENDPOINT, in demo_config.h."
```

Comment:
Will this error ever trigger? Because we defined this macro in the config if it wasn't defined already

Reply:
If the customer happens to remove it from the config file, then this failure would remind them instead.